doc: Add a known issues page (#2998)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

========================

.. rst-class:: v130

Build configuration issues
  The build configuration consisting of :ref:`bootloader`, :ref:`secure_partition_manager`, and application does not work.

  **Workaround:** Either include MCUboot in the build or use MCUboot instead of the immutable bootloader.

.. rst-class:: v130

Flash commands only program one core
  ``west flash`` and ``ninja flash`` only program one core, even if multiple cores are included in the build.

  **Workaround:** Execute the flash command from inside the build directory of the child image that is placed on the other core (for example, :file:`build/hci_rpmsg`).

Secure Partition Manager (SPM)
==============================

.. rst-class:: v130

NCSIDB-114: Default logging causes crash
  Enabling default logging in the :ref:`secure_partition_manager` sample makes it crash if the sample logs any data after the application has booted (for example, during a SecureFault, or in a secure service).
  At that point, RTC1 and UARTE0 are non-secure.

  **Workaround:** Do not enable logging and add a breakpoint in the fault handling, or try a different logging backend.

nRF9160
*******

Asset tracker
=============

> .. rst-class:: v131

Author: This probably applies to more releases.

Reviewer: Checking with @umapraseeda .

Reviewer: Info from Uma: add v130 tag as well, but no earlier ones.

  The behavior depends on the cellular network connection.

.. rst-class:: v120

GPS sockets and SUPL client library stops working
  The :ref:`gps_with_supl_support_sample` sample stops working if :ref:`supl_client` support is enabled, but the SUPL host name cannot be resolved.

  **Workaround:** Insert a delay (``k_sleep()``) of a few seconds after the ``printf`` on line 294 in :file:`main.c`.

.. rst-class:: v100 v110 v120

Calling nrf_connect immediately causes fail
  nrf_connect fails if called immediately after initialization of the device.
  A delay of 1000 ms is required for this to work as intended.

.. rst-class:: v030 v040 v100 v110 v120

Problems with RTT Viewer/Logger
  The SEGGER Control Block cannot be found by automatic search by the RTT Viewer/Logger.

  **Workaround:** Set the RTT Control Block address to 0 and it will try to search from address 0 and upwards.
  If this does not work, look in the ``builddir/zephyr/zephyr.map`` file to find the address of the ``_SEGGER_RTT`` symbol in the map file and use that as input to the viewer/logger.

.. rst-class:: v100 v110 v120 v130

> Large packets

Author: ```suggestion
Receive error with large packets
```

.. rst-class:: v110

:option:`CONFIG_NFCT_IRQ_PRIORITY` requires specific value
  :option:`CONFIG_NFCT_IRQ_PRIORITY` must be set to 5 or less when selecting :option:`CONFIG_BT_LL_SOFTDEVICE` on :ref:`peripheral_hids_keyboard`.

.. rst-class:: v110

Wait time required after a directed high duty cycle advertiser times out
  When selecting :option:`CONFIG_BT_LL_SOFTDEVICE`, if a directed high duty cycle advertiser times out, the application might have to wait a short time before starting a new connectable advertiser.
  Otherwise, starting the advertiser will fail.

.. rst-class:: v110

Assert risk after performing a DLE procedure
  The :ref:`nrfxlib:softdevice_controller` 0.3.0-3.prealpha might assert when receiving a packet with an CRC error on LE Coded PHY after performing a DLE procedure where RX Octets is changed to a value above 140.

Other issues
------------

.. rst-class:: v130

NCSDK-5711: High-throughput transmission can deadlock the receive thread
  High-throughput transmission can deadlock the receive thread if the connection is suddenly disconnected.

.. rst-class:: v130

> NCSDK-5711: High-throughput transmission can deadlock the transmit thread

Author: Duplicate of the one above?

    Bi-directional high-throughput traffic can deadlock the transmit thread.

.. rst-class:: v120

Only secure applications can use |BLE|
  Bluetooth LE cannot be used in a non-secure application, for example, an application built for the ``nrf5340_dk_nrf5340_cpuappns`` board.

  **Workaround:** Use the ``nrf5340_dk_nrf5340_cpuapp`` board instead.

.. rst-class:: v120

> Peripheral HIDS keyboard sample cannot be used with SoftDevice Controller

Author: ```suggestion
Peripheral HIDS keyboard sample cannot be used with nRF Bluetooth LE Controller
```
This one's tricky, because it's called SoftDevice Controller now (and thus links say SD Controller), but it was called nRF Bluetooth LE Controller at time of the release.

  The library uses the MPSL Clock driver, which does not provide an API for asynchronous clock operation.
  NFC requires this API to work correctly.

.. rst-class:: v120

Peripheral HIDS mouse sample advertising issues
  When the :ref:`peripheral_hids_mouse` sample is used with the Zephyr Bluetooth LE Controller, directed advertising does not time out and the regular advertising cannot be started.

.. rst-class:: v120

Central HIDS sample issues with directed advertising
  The :ref:`bluetooth_central_hids` sample cannot connect to a peripheral that uses directed advertising.

.. rst-class:: v110

Unstable samples
  Bluetooth Low Energy peripheral samples are unstable in some conditions (when pairing and bonding are performed and then disconnections/re-connections happen).

.. rst-class:: v110 v120

:option:`CONFIG_BT_SMP` alignment requirement
  When running the :ref:`bluetooth_central_dfu_smp` sample, the :option:`CONFIG_BT_SMP` configuration must be aligned between this sample and the Zephyr counterpart (:ref:`zephyr:smp_svr_sample`).
  However, security is not enabled by default in the Zephyr sample.

.. rst-class:: v100 v110 v120

> Reconnection issues on some OSes

Author: ```suggestion
Reconnection issues on some operating systems
```

>   On some operating systems, the nrf_desktop application is unable to reconnect to a host.

Author: link?

.. rst-class:: v100 v110

> `central_uart` cannot handle long strings

Author: Would be nice to link (also line 335).
